modules/parent-join/src/test/java/org/elasticsearch/join/query/HasChildQueryBuilderTests.java

```diff
@@ -174,19 +172,12 @@ protected void doAssertLuceneQuery(HasChildQueryBuilder queryBuilder, Query quer
         if (queryBuilder.innerHit() != null) {
             // have to rewrite again because the provided queryBuilder hasn't been rewritten (directly returned from
             // doCreateTestQueryBuilder)
-            queryBuilder = (HasChildQueryBuilder) queryBuilder.rewrite(searchContext.getQueryShardContext());
+            queryBuilder = (HasChildQueryBuilder) queryBuilder.rewrite(context);
             Map<String, InnerHitContextBuilder> innerHitBuilders = new HashMap<>();
             InnerHitContextBuilder.extractInnerHits(queryBuilder, innerHitBuilders);
-            final InnerHitsContext innerHitsContext = new InnerHitsContext();
-            for (InnerHitContextBuilder builder : innerHitBuilders.values()) {
-                builder.build(searchContext, innerHitsContext);
-            }
```

Comment:
Is the plan to pass a QueryShardContext here as well eventually?

Reply:
No, this is now done in the `FetchSubPhase` and the `searchContext` is required since we need to clone it.